### PR TITLE
Add Nats Streaming (aka stan) support in Keda 

### PR DESCRIPTION
**Pull request by balchua**
Just wondering if this is something the community is interested with.

### CODE REVIEW

File: spec/triggers/stan.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# How to test the Nats streaming scaler
```
Comment (tomkerkhove) [inline spec/triggers/stan.md]: This is not really the best place for this, maybe move this to another location?

@ahmelsayed might be able to help you with this

File: spec/triggers/stan.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+## Start a Nats Streaming publisher
+
+See the example [publisher](https://github.com/balchua/gonuts/tree/master/pub) code.
+
+## Start a Nats Streaming consumer
+
+See the example [consumer](https://github.com/balchua/gonuts/tree/master/sub) code.
+
+
+## Apply `stan_scaledobject.yaml`
```
Comment (tomkerkhove) [inline spec/triggers/stan.md]: Thanks for adding this, can you align the approach a bit with the others please?

Example: https://github.com/kedacore/keda/blob/master/spec/triggers/azure-service-bus.md

Comment (balchua) [reply spec/triggers/stan.md]: Thanks for the review.  Can samples go to `keda/samples`?

Comment (tomkerkhove) [reply spec/triggers/stan.md]: If you have a look at the example you'll see that only an example of the trigger itself is required here, full examples should go to /examples like you did in this PR

**Comment by balchua** [comment]
I've moved the documentation along with the rest of the scalers. 
Do let me know if there are other things that aren't aligned with the rest of the scalers.

Thanks again for the review.

**Pull request merged**